src/main/java/com/suse/saltstack/netapi/client/SaltStackClient.java

```diff
@@ -97,14 +97,18 @@ public Token login(String username, String password)
      * @return authentication token as {@link Token}
      * @throws SaltStackException if anything goes wrong
      */
-    public Token login(String username, String password, String eauth)
+    public Token login(final String username, final String password, final String eauth)
             throws SaltStackException {
-        JsonObject json = new JsonObject();
-        json.addProperty("username", username);
-        json.addProperty("password", password);
-        json.addProperty("eauth", eauth);
+        Map<String, String> kwargs = new LinkedHashMap() {
```

Comment:
Right, that's what I was commenting - making them final could prevent some boilerplate in #28.

The only issue with final arguments in this PR case is IMHO that it makes the method signatures even loooonger. If you think this is a problem, I'll revert back to non-final args (but will have to initialize the map the "traditional" way, i.e. without "double braces" syntax).